src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -1,410 +0,0 @@
-        private const string ConfigFileSuffix = ".eventpipeconfig";
-        private const int EnabledPollingIntervalMilliseconds = 1000; // 1 second
-        private const int DisabledPollingIntervalMilliseconds = 20000; // 20 seconds
-        private const uint DefaultCircularBufferMB = 1024; // 1 GB
-        private const char ConfigEntryDelimiter = '=';
-        private const char ProviderConfigDelimiter = ',';
-        private const char ConfigComponentDelimiter = ':';
-
-        // Config file keys.
-        private const string ConfigKey_Providers = "Providers";
-        private const string ConfigKey_CircularMB = "CircularMB";
-        private const string ConfigKey_OutputPath = "OutputPath";
-        private const string ConfigKey_ProcessID = "ProcessID";
-
-        // The default set of providers/keywords/levels.  Used if an alternative configuration is not specified.
-        private static EventPipeProviderConfiguration[] DefaultProviderConfiguration => new EventPipeProviderConfiguration[]
-        {
-            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntime", 0x4c14fccbd, 5, null),
-            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntimePrivate", 0x4002000b, 5, null),
-            new EventPipeProviderConfiguration("Microsoft-DotNETCore-SampleProfiler", 0x0, 5, null),
-        };
-
-        // Singleton controller instance.
-        private static EventPipeController? s_controllerInstance;
-
-        // Controller object state.
-        private Timer m_timer;
-        private string m_configFilePath;
-        private bool m_configFileExists;
-
-        // Initialization flag used to avoid initializing FrameworkEventSource on the startup path.
-        internal static bool Initializing { get; private set; }
-
-        internal static void Initialize()
-        {
-            // Don't allow failures to propagate upstream.  Ensure program correctness without tracing.
-            try
-            {
-                Initializing = true;
-
-                if (s_controllerInstance == null)
-                {
-                    int enabled = Config_EnableEventPipe;
-                    if (enabled > 0)
-                    {
-                        // Enable tracing immediately.
-                        // It will be disabled automatically on shutdown.
-                        EventPipe.Enable(BuildConfigFromEnvironment());
-                    }
-                    // If not set at all, we listen for changes in the control file.
-                    else if (enabled != 0)
-                    {
-                        // Create a new controller to listen for commands.
-                        s_controllerInstance = new EventPipeController();
-                    }
-                    // If enable is explicitly set to 0, then don't start the controller (to avoid overhead).
-                    RuntimeEventSource.Initialize();
```

Comment:
You'll need to move this code to StartupHookProvider as this is the EventSource with runtime event counters.